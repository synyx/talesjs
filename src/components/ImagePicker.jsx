import React from "react";
import PropTypes from "prop-types";

export default class ImagePicker extends React.Component {
  static propTypes = {
    onFileSelect: PropTypes.func.isRequired,
  };

  handleFileSelect(event) {
    const file = event.target.files[0];
    this.props.onFileSelect(file);
  }

  render() {
    return (
      <input
        type="file"
        accept="image/*"
        onChange={event => this.handleFileSelect(event)}
      />
    );
  }
}
