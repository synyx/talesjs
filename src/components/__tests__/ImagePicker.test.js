import React from "react";
import ImagePicker from "../ImagePicker";
import { renderShallow, renderIntoDocument } from "../../../test";

describe("ImagePicker", () => {
  beforeEach(() => {});

  it("renders", () => {
    const props = setupProperties();
    const tree = renderShallow(<ImagePicker {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("invokes #onFileSelect on file selection", () => {
    const props = setupProperties();
    const { simulate } = renderIntoDocument(<ImagePicker {...props} />);

    expect(props.onFileSelect).not.toHaveBeenCalled();

    const file = { type: "image/foo" };
    simulate.change("input", { target: { files: [file] } });

    expect(props.onFileSelect).toHaveBeenCalledWith(file);
  });

  function setupProperties() {
    return {
      onFileSelect: jest.fn(),
    };
  }
});
