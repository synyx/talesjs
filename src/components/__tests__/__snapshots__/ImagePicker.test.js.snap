// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImagePicker renders 1`] = `
<input
  accept="image/*"
  onChange={[Function]}
  type="file"
/>
`;
